rskj-core/src/main/java/org/ethereum/db/IndexedBlockStore.java

```diff
@@ -510,7 +510,10 @@ private static BlockInfo getBlockInfoForHash(List<BlockInfo> blocks, byte[] hash
             byte[] hash = blockInfo.getHash().getBytes();
             Block block = getBlockByHash(hash);
 
-            result.add(block);
+            // TODO(mc) investigate and fix this, probably a cache invalidation problem
+            if (block != null) {
+                result.add(block);
+            }
```

Comment:
We should fix the whole snapshot issue, this should be better than the current implementation